packages/babel-plugin-proposal-dynamic-import/test/fixtures/amd/no-interop/output.js

```diff
@@ -1,3 +1,3 @@
 define(["require"], function (_require) {
-  var modP = new Promise((_resolve, _reject) => _require(["mod"], imported => _resolve(imported), _reject));
+  var modP = Promise.resolve().then(() => new Promise((_resolve, _reject) => _require(["mod"], imported => _resolve(imported), _reject)));
```

Comment:
Since the AMD `require` function is already asynchronous, we don't need to defer it to `.then` — the original code was fine.

Reply:
Ok, I just need to set the `buildDynamicImport`'s `runInThen` to `false` for it.

https://github.com/babel/babel/blob/75996fe68e4d00aed78e465f330fcb37a406d71f/packages/babel-plugin-transform-modules-amd/src/index.ts#L102-L114

Reply:
Now the logic looks like this:

CommonJS (Deferred to then):
https://github.com/babel/babel/blob/9b2100566283b50c9ed895d03ee1f1c0a7bb93c0/packages/babel-helper-module-transforms/src/dynamic-import.ts#L44-L50

SystemJS (Needs to return a rejected promise so wrap it in a promise):
https://github.com/babel/babel/blob/9b2100566283b50c9ed895d03ee1f1c0a7bb93c0/packages/babel-helper-module-transforms/src/dynamic-import.ts#L51-L56

AMD (Builder already returns a `new Promise` so do nothing):
https://github.com/babel/babel/blob/9b2100566283b50c9ed895d03ee1f1c0a7bb93c0/packages/babel-helper-module-transforms/src/dynamic-import.ts#L57-L60